synapse/replication/http/_base.py

```diff
@@ -220,6 +220,9 @@ async def send_request(
                 if instance_name == local_instance_name:
                     raise Exception("Trying to send HTTP request to self")
                 if instance_name not in instance_map:
+                    # TODO: remove below condition after master officially becomes main
+                    if instance_name == "master":
+                        instance_name = "main"
```

Comment:
It's probably easiest to debug problems if the worker instance matched what was in the config.

What's stopping us from changing the default config in a separate PR instead?

Reply:
That is a much bigger project, unfortunately. The main process has been named `master` for a long time, and references to it as such are peppered throughout the code base. IIRC, in order to change just this one part, would involve 3 other(seemingly unrelated) code spots. It is on my agenda to try and get that done.

That being said, in this specific case, it allows the `Exception` to error with a more intuitive `Instance main not in instance_map config`, which is what it should have been(This was a drive by fix that got neglected in it's original PR). There were several reports(leading to a fix up and point version release) where this was reported, and it lead to some confusion. This was to help clear that up, as consistency in error reporting. Does that make sense?

Reply:
I'm haven't delved into or used the worker config to have an informed opinion here but naively, it seems like the mismatch wouldn't be more intuitive/clear. It's literally saying "x" isn't in `instance_map` config.

Reply:
I'm not sure I completely follow. The error message (before this) said that `Instance master not in instance_map config`, which was a minor problem as `master` wasn't supposed to be on the `instance_map` at all, but `main` was(as of the [PR](https://github.com/matrix-org/synapse/pull/15491) that brought that change in so we could get rid of the various `worker_replication_*` style of configs). Until the name `master` can be removed from the Synapse code base when referencing the main process, this just helped make the error message clearer.

Reply:
I kinda understand better after stumbling onto [`docs/usage/configuration/config_documentation.md#instance_map`](https://github.com/matrix-org/synapse/blob/1404f68a03e684434dea6b2a9a5c2739c719549d/docs/usage/configuration/config_documentation.md#instance_map) which mentions `main`.

I don't know where the translation of `main`/`master` in Synapse happens.

Perhaps this is just best split out to a separate PR as it looks adjacent to the unix socket changes :fast_forward:

Reply:
That's fine. I can do that.

Reply:
Removed in [eb65470](https://github.com/matrix-org/synapse/pull/15708/commits/eb65470718333478f1eb68c155fab7318cb68fa4)